tests/app/main/views/test_index.py

```diff
@@ -43,16 +43,20 @@ def it_should_show_future_events_in_cards(self, client, sample_future_events, sa
 
         assert content == intro_course['title']
 
-    def it_should_show_past_and_future_events_in_cards(self, client, sample_articles_summary, sample_all_events):
+    def it_should_show_past_and_future_events_in_cards(
+        self, mocker, client, sample_articles_summary, sample_future_event_for_cards, sample_past_events_for_cards
+    ):
         response = client.get(url_for(
             'main.index'
         ))
         page = BeautifulSoup(response.data.decode('utf-8'), 'html.parser')
-        content = page.find("div", {"class": "past_corner"}).string
+        mocker.patch('app.main.views.index.api_client.get_events_in_future', sample_future_event_for_cards)
+        mocker.patch('app.main.views.index.api_client.get_events_past_year', sample_past_events_for_cards)
 
-        past_event = [e for e in sample_all_events if e['past'] == 'Past Event'][0]
+        content = page.find_all(id="past_event")
+        print(page)
 
-        assert content == past_event['past']
+        assert len(content) == 2
```

Comment:
For some reason this is displaying all 4 past events which ive listed in the confest.py and im not sure why. Its not even showing the one future event im mocking. And ideas? @kenlt-uk

Reply:
Please read the commit message which should fix the tests, once you have understood them please clean up the commits.